File: codeforces/34B - Sale/a.py
def main():
    _, m = map(int, input().split())
    a = list(map(int, input().split()))
    a.sort()
    res = 0
    for x in a:
        if x >= 0 or m == 0:
            break
        else:
            res = res + (-x)
        m -= 1
    print(res)

if __name__ ==  "__main__":
    main()

File: codeforces/34B - Sale/a.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);
  int n, m;
  cin >> n >> m;
  vector<int> a;
  for(int i = 0; i < n; i++) {
    int x;
    cin >> x;
    a.push_back(x);
  }
  sort(a.begin(), a.end());
  int res = 0;
  for(auto x: a) {
    if(x >= 0 || m == 0) {
      break;
    } else {
      res += (-x);
      m--;
    }
  }
  cout << res;
  return 0;
}
